### ISSUE TITLE
Feature request: Allow dart formatter to sort lines in defined area of code

### ISSUE THREAD

**Issue by fniemann** [dart-lang/dart_style#754]
I think it could be useful to be able to define an area of code that should be kept sorted by dartfmt.
This would be useful for any larger list of items that should stay sorted for better readability and understanding.

This could be as simple as two comments defining the start and end of the sorted area, e.g.
var listOfStrings = [
  // START sorting
  'a',
  'b',
  'c',
  // END sorting
];

It could even be nice to allow different kinds of sorting, e.g.
// START sorting: dictionary
// START sorting: dictionary reverse
// START sorting: dictionary ignore-case

**Comment by natebosch** [comment]
Dartfmt doesn't touch ordering, and I think that's a good thing.

This seems like a problem better solved by your text editor than by dartfmt...

**Comment by fniemann** [comment]
Yes, a text editor can solve this problem for me. But it won't solve it in a shared code environment.
The dart formatter would be able to enforce it, e.g. in a presubmit scenario.

Wrt not touching ordering: That's why I proposed the comments, to explicitly call out a section where it would be allowed to touch ordering. Obviously, it shouldn't do that by default on any list.

**Comment by parren-google** [comment]
Note that the sorting is for easy scanning when reading the code, often in a read-only environment. Getting that without burdening writers and code reviewers seems like a simple win here.
